02_bubble/joel00wood/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+)
+
+func bubble(s []int) []int {
+	c := make([]int, len(s))
+	copy(c, s)
+	var swapped bool
+	for {
+		for i := 1; i < len(c); i++ {
+			if c[i-1] > c[i] {
+				c[i-1], c[i] = c[i], c[i-1]
+				swapped = true
+			}
+		}
+		if !swapped {
+			break
+		} else {
+			swapped = false
+		}
+	}
+	return c
+}
+
+func insertion(s []int) []int {
+	c := make([]int, len(s))
+	copy(c, s)
+	for i := 1; i < len(s); i++ {
+		for j := i; j > 0; j-- {
+			if c[j-1] > c[j] {
+				c[j-1], c[j] = c[j], c[j-1]
+			}
+		}
+	}
+	return c
+}
+
+func mergeSort(s []int) []int {
```

Comment:
this function will modify source slice. please copy input like in `insertion` (line: 29)

Reply:
Updated as described